scripts/persona_driven_data_gen/README.md

```diff
@@ -0,0 +1,48 @@
+To start make sure you have your OpenAI and Anthropic API keys and have installed the libraries listed in `requirements.txt`:
+
+```
+pip install -r requirements.txt
+```
+
+This folder contains code to synthetically generate data (both prompts and responses) for target skill using a [persona-driven approach](https://arxiv.org/pdf/2406.20094):
+
+
+**1- Precide Instruction Following:**
+
+```
+# Generate Instruction Following prompts
+python persona_driven_generate_ifdata.py --model "gpt-4o" --start_index 0 --end_index <SAMPLE_SIZE> --output_path if_prompts.jsonl --openai_key Z --org_id YYY --dataset ai2-adapt-dev/personahub_personas --template instruction_following
+
+# Generate Responses for generated prompts
+python persona_driven_generate_ifdata.py --model "gpt-4o" --start_index 0 --end_index <SAMPLE_SIZE> --output_path if_solutions.jsonl --openai_key Z --org_id YYY --dataset if_prompts.jsonl --template instruction_following_solution
+
+# Rewrite prompts to form Rejected Response (used for Presona-IF DPO data)
+python persona_driven_generate_ifdata.py --model "gpt-4o" --start_index 0 --end_index <SAMPLE_SIZE> --output_path if_solutions.jsonl --openai_key Z --org_id YYY --dataset if_prompts.jsonl --template rewrite_if_prompt
```

Comment:
Maybe include some example outputs to give the readers on what to expect after running the script. Kind of like how I included `Here is an example created dataset: https://huggingface.co/datasets/vwxyzjn/rejection_sampling_1727887719` in  https://github.com/allenai/open-instruct/blob/main/docs/algorithms/rejection_sampling.md#scoring-completions